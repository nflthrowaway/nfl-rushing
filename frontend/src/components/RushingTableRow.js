import React from "react";

import { columns, columnToMetricMap, metricToColumnMap } from '../common/constants';

const RushingTableRow = ({ row, index }) => {
    return (
        <tbody>
            <tr key={index} className={'RushingTableRow'} data-testid={'table-row'}>
                {columns.reduce((result, column, index) => {
                    if (column === metricToColumnMap.LongestRushTD) {
                        return result;
                    }

                    let cellData = row[columnToMetricMap[column]];
                    if (column === metricToColumnMap.LongestRush && row['LongestRushTD']) {
                        cellData += 'T';
                    }
                    result.push((
                        <td className={'RushingTableCell'} key={index}>
                            {cellData}
                        </td>
                    ));
                    return result;
                }, [])}
            </tr>
        </tbody>
    )
}

export default RushingTableRow;
