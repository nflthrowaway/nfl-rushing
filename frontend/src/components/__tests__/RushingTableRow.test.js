import { render } from '@testing-library/react';
import RushingTableRow from '../RushingTableRow';

test('renders row', async () => {
    const testData = {
        Player: 'Name',
        Team: 'Team',
        Position: 'Position',
        RushingAttempts: 2,
        RushingAttemptsPerGame: 1,
        TotalRushingYards: 20,
        RushingAverageYardsPerAttempt: 10,
        RushingYardsPerGame: 20,
        TotalRushingTouchdowns: 0,
        LongestRush: 10,
        LongestRushTD: false,
        RushingFirstDowns: 0,
        RushingFirstDownPercentage: 0,
        Rushing20PlusYardsEach: 0,
        Rushing40PlusYardsEach: 0,
        RushingFumbles: 0
    };

    const { findByText, getByTestId } = render(<RushingTableRow row={testData} />);
    await findByText('Name');
    const table = getByTestId('table-row');
    expect(table).toMatchSnapshot();
});