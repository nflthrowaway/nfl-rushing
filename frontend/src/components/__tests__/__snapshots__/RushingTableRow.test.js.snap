// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders row 1`] = `
<tr
  class="RushingTableRow"
  data-testid="table-row"
>
  <td
    class="RushingTableCell"
  >
    Name
  </td>
  <td
    class="RushingTableCell"
  >
    Team
  </td>
  <td
    class="RushingTableCell"
  >
    Position
  </td>
  <td
    class="RushingTableCell"
  >
    2
  </td>
  <td
    class="RushingTableCell"
  >
    1
  </td>
  <td
    class="RushingTableCell"
  >
    20
  </td>
  <td
    class="RushingTableCell"
  >
    10
  </td>
  <td
    class="RushingTableCell"
  >
    20
  </td>
  <td
    class="RushingTableCell"
  >
    0
  </td>
  <td
    class="RushingTableCell"
  >
    10
  </td>
  <td
    class="RushingTableCell"
  >
    0
  </td>
  <td
    class="RushingTableCell"
  >
    0
  </td>
  <td
    class="RushingTableCell"
  >
    0
  </td>
  <td
    class="RushingTableCell"
  >
    0
  </td>
  <td
    class="RushingTableCell"
  >
    0
  </td>
</tr>
`;
